.github/workflows/python_package.yml

```
@@ -113,12 +113,19 @@ jobs:
             --rm \
             -v $(pwd):/opt/lgb-build \
             -w /opt/lgb-build \
-            python:3.12 \
+            python:3.13 \
             /bin/bash ./.ci/test-python-latest.sh
-  test-oldest-versions:
+  test-old-versions:
     name: Python - oldest supported versions (ubuntu-latest)
     runs-on: ubuntu-latest
     timeout-minutes: 60
+    strategy:
+      fail-fast: false
+      matrix:
+        # end-of-life Python versions
+        python_version:
+          - 3.7
+          - 3.8
```

To avoid potential floating-point manipulation like `3.10` -> `3.1`
```suggestion
          - '3.7'
          - '3.8'
```

---

Agree, let's do that.